### PR TITLE
[DevOps] Snyk Integration

### PR DESCRIPTION
**Pull request by ChathurindaRanasinghe**
Integrated Snyk [issue #241](https://github.com/h2oai/h2o-ops/issues/241)

### CODE REVIEW

File: .github/workflows/snyk-scan.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Snyk Security Vulnerability Scan
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+  push:
+    tags:
+      - 'v.[0-9]+.[0-100]+.[0-100]+'
```
Comment (EshamAaqib) [inline .github/workflows/snyk-scan.yml:8]: Just wondering, What if we add `[0-100]` instead of `[0-9]`. WDYT @hasithjp ?

**Review by EshamAaqib — APPROVED** [review]
LGTM

**Comment by ChathurindaRanasinghe** [comment]
image
https://github.com/h2oai/wave-ml/actions/runs/4281132589/jobs/7453771657#step:6:36

@ledell I think it's better to update `sklearn` in requirements files to `scikit-learn`. Seems like `sklearn` package is deprecated.

**Review by ledell — APPROVED** [review]
Looks good to me.

**Review by hasithjp — APPROVED** [review]

**Pull request merged**